app/main.py

```diff
@@ -1,2 +1,48 @@
 class Dictionary:
-    pass
+    def __init__(self):
+        self.capacity = 0
+        self.my_dict_length = 8
+        self.my_dict = [[] for _ in range(self.my_dict_length)]
+
+    def __setitem__(self, key, value):
+        hash_ = hash(key)
+        index = hash_ % self.my_dict_length
+        if not self.my_dict[index]:
+            self.my_dict[index] = [hash_, key, value]
+            self.capacity += 1
+        elif self.my_dict[index][1] == key:
+            self.my_dict[index][2] = value
+        else:
+            index = (index + 1) % self.my_dict_length
+            while True:
+                if not self.my_dict[index]:
+                    self.my_dict[index] = [hash_, key, value]
+                    self.capacity += 1
+                    break
+                if self.my_dict[index][1] == key:
+                    self.my_dict[index][2] = value
+                    break
```

Comment:
You did the same check (10-14 and 18-24) twice. You can write it concisely within one loop.